### ISSUE TITLE
Set torch.nn.utils.clip_grad_norm_(error_if_nonfinite=True) if Trainer.terminate_on_nan is True

### ISSUE THREAD

**Issue by cowwoc** [PyTorchLightning/pytorch-lightning#9794]
Currently, if a user sets Trainer `gradient_clip_val` to a non-zero value and `terminate_on_nan=True` then `PrecisionPlugin.clip_grad_by_norm()` invokes `torch.nn.utils.clip_grad_norm_(parameters, clip_val)` without setting `error_if_nonfinite=True`.

It stands to reason that if the user set `terminate_on_nan=True` then he wants us to set `error_if_nonfinite=True` when invoking `torch.nn.utils.clip_grad_norm_()`.

cc @borda

**Comment by ananthsub** [comment]
@cowwoc - one proposal has been to use torch.autograd.detect_anomaly instead of lightning's terminate on nan: #8313

do you know if this context applied over fitting would automatically cover this case of nan gradients?

**Comment by cowwoc** [comment]
I'm fine with unifying this with `torch.autograd.detect_anomaly`. We could have a single debug flag that triggers all these checks at once.

> do you know if this context applied over fitting would automatically cover this case of nan gradients?

Sorry, I don't understand the question.

**Comment by cowwoc** [comment]
@ananthsub Still waiting on your clarification.
